pinot-batch-ingestion/pinot-standalone/src/main/java/org/apache/pinot/ingestion/standalone/SegmentGenerationJobRunner.java

```diff
@@ -227,41 +236,83 @@ public void run()
     }
   }
 
-  private Schema getSchema()
-      throws Exception {
-    URI schemaURI = new URI(_spec.getTableSpec().getSchemaURI());
+  private Schema getSchema() {
+    URI schemaURI;
+    try {
+      schemaURI = new URI(_spec.getTableSpec().getSchemaURI());
+    } catch (URISyntaxException e) {
+      throw new RuntimeException("Schema URI is not valid - " + _spec.getTableSpec().getSchemaURI(), e);
+    }
     String scheme = schemaURI.getScheme();
     String schemaJson;
     if (PinotFSFactory.isSchemeSupported(scheme)) {
       // Try to use PinotFS to read schema URI
       PinotFS pinotFS = PinotFSFactory.create(scheme);
-      InputStream schemaStream = pinotFS.open(schemaURI);
-      schemaJson = IOUtils.toString(schemaStream, StandardCharsets.UTF_8);
+      InputStream schemaStream;
+      try {
+        schemaStream = pinotFS.open(schemaURI);
+      } catch (IOException e) {
+        throw new RuntimeException("Failed to open schema file stream on Pinot fs - " + schemaURI, e);
```

Comment:
Suggeste: "Failed to fetch schema from PinotFS". Maybe also useful to add the pinot-fs name here.

Reply:
pinot-fs name  is from the uri scheme, so scheme uri should provide enough information to debug.